### DISCUSSION TITLE
Upgrade from hypfer, won't save maps

### DISCUSSION THREAD

**Discussion by oddomatik** [rand256/valetudo#514] [General]
I haven't dug too deep into source yet, but it appears that after moving from hypfer->rand256 I am unable to save or zones. Anyone knowledgeable in that area able to tell me how the filesystem gets mounted up, what permissions it should have, etc? I believe I initially backed up and zero'd out some of the map files as an attempt to get it working. What is the purpose of all the different map files, and how can I debug this?

Thanks in advance for any assistance.

**Comment by sca075** [comment]
@oddomatik I did had the same doubts after the switch from to RE.
I do have an S5 and you need to enable the "experimental" future on the vacuum options and send the robot around (clean/scan) for two times. After that is possible to edit the "zones" 
If you have a gen1 I don't know if the procedure works.. 
I did anyhow work out a camera that can extract (home assistant wise) the map "valetudo vacuum camera" this could help you to know if the segments are in the vacuum or not

**Reply by oddomatik** [reply]
Thank you for the response. I don't think I've tried doing a map run twice before cleaning, perhaps that's documented and I overlooked it.

**Reply by sca075** [reply]
I just had a look on my vacuum "persisting data" on System Settings should be set ON hope this helps

**Reply by oddomatik** [reply]
Thanks! In my case, it is. I do see a map, but whenever I try to set zones or no-go zones, they are forgotten

**Reply by sca075** [reply]
what vacuum do you use? the V1 have no zones so far I know.. probably this is the "issue" ;)

**Reply by oddomatik** [reply]
It's an S7 - I'll do some further testing/digging and come back with thorough information - I was mainly curious if I had missed something obvious or if it was a known issue before spending the time - which is me being lazy. I appreciate your help!

**Reply by oddomatik** [reply]
The main issue is that while those settings are enabled, no maps actually get saved. Just whatever was from the previous run.
When I try to edit forbidden zones, I get Error 500 - Current map is null.

**Comment by pidator** [comment]
> It's an S7

The development of this fork based mainly on the S5, you are the first one I read about running it on an S7. There's an high probability that the features you are trying to use simply don't work.
